pkg/client/intermediate/intermediate.go

```diff
@@ -0,0 +1,139 @@
+package intermediate
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"sort"
+
+	"github.com/bitnami-labs/charts-syncer/pkg/client"
+
+	"github.com/juju/errors"
+
+	"github.com/bitnami-labs/charts-syncer/internal/utils"
+	"github.com/bitnami-labs/charts-syncer/pkg/client/types"
+	"helm.sh/helm/v3/pkg/chart"
+)
+
+var (
+	versionRe = regexp.MustCompile("(.*)-(\\d+\\.\\d+\\.\\d+)\\.bundle.tar")
+)
+
+// Repo allows to operate a chart repository.
+type Repo struct {
+	dir     string
+	entries map[string][]string
+}
+
+// NewIntermediateClient returns a ReadWriter object
+var NewIntermediateClient = func(intermediateBundlesPath string) (client.ReadWriter, error) {
+	return New(intermediateBundlesPath)
+}
+
+// New creates a Repo object from an api.Repo object.
+func New(dir string) (*Repo, error) {
```

Comment:
it's interesting that the entries get loaded in the constructor instead of the list method which to me makes more sense. In fact, the lest logic a constructor has the better.

To me, entries should be added when calling List similar to what a DB will do while getting queried

Reply:
+1

Reply:
There are other methods that relies on the `entries` info initialised in the constructor (ListChartVersions(), Has(), Upload())...moving the entries initialisation to the List() method means that we cannot call to the other methods before List() because entries would be empty or we will need to retrieve the info always from the List() method.

The later means that we would be reading the content of a folder several times per chart while having the entries populated just once seems more efficient to me. I would leave it as it is